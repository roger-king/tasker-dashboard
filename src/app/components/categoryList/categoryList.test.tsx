import React from 'react';
import renderer from 'react-test-renderer';
import CategoryList from './index';

describe('<CategoryList />', () => {
    const categories = ['All', 'main', 'create'];
    let currentCategory: string | null = null;
    it('Matches the snapshot with ALL selected', () => {
        const select = jest.fn();
        const openModal = jest.fn();
        const list = renderer
            .create(
                <CategoryList
                    categories={categories}
                    current={currentCategory}
                    selectCategory={select}
                    openModal={openModal}
                />,
            )
            .toJSON();
        expect(list).toMatchSnapshot();
    });

    it('Matches the snapshot with MAIN selected', () => {
        currentCategory = categories[1];
        const select = jest.fn();
        const openModal = jest.fn();
        const list = renderer
            .create(
                <CategoryList
                    categories={categories}
                    current={currentCategory}
                    selectCategory={select}
                    openModal={openModal}
                />,
            )
            .toJSON();
        expect(list).toMatchSnapshot();
    });
});
