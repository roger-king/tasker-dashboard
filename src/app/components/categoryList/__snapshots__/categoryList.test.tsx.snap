// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryList /> Matches the snapshot with ALL selected 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 eSkhpc sc-bdVaJa bDWFJH"
>
  <button
    className="StyledButton-sc-323bzc-0 hsEINM"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 jHQJnz"
    >
      <svg
        aria-label="Add"
        className="StyledIcon-ofa7kd-0 fIBwIl"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,22 L12,2 M2,12 L22,12"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
      />
      Add
    </div>
  </button>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 eTnusT"
    id="category-all-menu-item"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 gcukze"
    >
      All topics
    </h4>
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
    id="category-All-menu-item"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 gcukze"
    >
      All
    </h4>
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
    id="category-main-menu-item"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 gcukze"
    >
      main
    </h4>
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
    id="category-create-menu-item"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 gcukze"
    >
      create
    </h4>
  </div>
</div>
`;

exports[`<CategoryList /> Matches the snapshot with MAIN selected 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 eSkhpc sc-bdVaJa bDWFJH"
>
  <button
    className="StyledButton-sc-323bzc-0 hsEINM"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 jHQJnz"
    >
      <svg
        aria-label="Add"
        className="StyledIcon-ofa7kd-0 fIBwIl"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,22 L12,2 M2,12 L22,12"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
      />
      Add
    </div>
  </button>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
    id="category-all-menu-item"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 gcukze"
    >
      All topics
    </h4>
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
    id="category-All-menu-item"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 gcukze"
    >
      All
    </h4>
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 eTnusT"
    id="category-main-menu-item"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 gcukze"
    >
      main
    </h4>
  </div>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
    id="category-create-menu-item"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <h4
      className="StyledHeading-sc-1rdh4aw-0 gcukze"
    >
      create
    </h4>
  </div>
</div>
`;
